activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -164,20 +164,35 @@ def drop_current(environment = env)
       end
 
       def migrate
-        raise "Empty VERSION provided" if ENV["VERSION"] && ENV["VERSION"].empty?
+        check_target_version
 
         verbose = ENV["VERBOSE"] ? ENV["VERBOSE"] != "false" : true
-        version = ENV["VERSION"] ? ENV["VERSION"].to_i : nil
         scope = ENV["SCOPE"]
         verbose_was, Migration.verbose = Migration.verbose, verbose
-        Migrator.migrate(migrations_paths, version) do |migration|
+        Migrator.migrate(migrations_paths, target_version) do |migration|
           scope.blank? || scope == migration.scope
         end
         ActiveRecord::Base.clear_cache!
       ensure
         Migration.verbose = verbose_was
       end
 
+      def check_target_version
+        if ENV["VERSION"] && ENV["VERSION"].empty?
+          raise "Empty VERSION provided"
```

Comment:
Checking for empties of `VERSION` is added by https://github.com/rails/rails/pull/28485 because `db:migrate:redo`, `db:migrate:up`, `db:migrate:down` require `VERSION`,
but `db:migrate` doesn't require  `VERSION`, so I think it `db:migrate` with empty `VERSION` should behave as without `VERSION`.